packages/sync-service/lib/electric/shape_cache.ex

```diff
@@ -209,9 +212,17 @@ defmodule Electric.ShapeCache do
         recover_shapes(state)
       end
 
-    # ensure publication filters are in line with existing shapes
+    # ensure publication filters are in line with existing shapes,
+    # and clean up cache if publication fails to update
     {publication_manager, publication_manager_opts} = opts.publication_manager
-    publication_manager.refresh_publication(publication_manager_opts)
+
+    try do
+      :ok = publication_manager.refresh_publication(publication_manager_opts)
+    catch
+      kind, reason when kind in [:exit, :error] ->
+        clean_up_all_shapes(state)
+        exit(reason)
+    end
```

Comment:
I don't think we should be converting exits to errors. I think a cleaner approach would be to add a separate `rescue` clause:


```suggestion
    rescue
      error -> 
        clean_up_all_shapes(state)
        reraise error, __STACKTRACE__
    catch
      :exit, reason ->
        clean_up_all_shapes(state)
        exit(reason)
    end
```